TMAP8 fails to build on Mac with new conda compilers

## Bug Description
As reported by @gsgall, TMAP8 fails to build when using the new conda compilers on Mac that were introduced in idaholab/moose#31935. When built (in this example, via SALAMANDER), the compiler errors with the output:

```
In file included from /Users/graysongall/projects/TMAP8/build/unity_src/interfacekernels_Unity.C:3:
/Users/graysongall/projects/TMAP8/src/interfacekernels/InterfaceSorption.C:83:20: error: a template argument list is expected after a name prefixed by the template keyword [-Wmissing-template-arg-list-after-template-kw]
   83 |     this->template paramError(
      |                    ^
Compiling C++ (in opt mode) /Users/graysongall/projects/TMAP8/test/src/base/TMAP8TestApp.C...
1 error generated.
make: *** [/Users/graysongall/projects/TMAP8/moose/framework/build.mk:167: /Users/graysongall/projects/TMAP8/build/unity_src/interfacekernels_Unity.aarch64-apple-darwin20.0.0.opt.lo] Error 1
```

As `paramError` isn't a templated method, [this line](https://github.com/idaholab/TMAP8/blob/50484fa4d223ef774addf32a9e219c5c27f0c8ae/src/interfacekernels/InterfaceSorption.C#L83C5-L83C19) should be fixed up.

## Steps to Reproduce

1. Create conda environment using `moose-dev=2026.01.27`.
2. Pull TMAP8 devel as of this issue.
3. Attempt to build TMAP8.

## Impact
Prevents use of TMAP8 on Mac platforms.

--- a/src/interfacekernels/InterfaceSorption.C
+++ b/src/interfacekernels/InterfaceSorption.C
@@ -80,8 +80,8 @@ InterfaceSorptionTempl<is_ad>::InterfaceSorptionTempl(const InputParameters & pa
             : nullptr)
 {
   if (!_use_flux_penalty && parameters.isParamSetByUser("flux_penalty"))
-    this->template paramError(
-        "flux_penalty", "The flux penalty should be specified only when use_flux_penalty = true");
+    this->paramError("flux_penalty",
+                     "The flux penalty should be specified only when use_flux_penalty = true");
 }
 
 template <bool is_ad>

